### ISSUE TITLE
Linear Layers on Graphs

### ISSUE THREAD

**Issue by aykroyd-id** [pyg-team/pytorch_geometric#405]
## ❓ Questions & Help
Hi,

I am simply trying to add a Linear layer at the end of a convolutional encoder, but I as soon as I add this layer my model stops training.

So my Encoder output is a graph of shape (B * N, C), where B is batch_size, N and C are nodes and channels that are the same along each graph.

I have:
```
self.flatten = lambda x : x.flatten().reshape(-1, N * C) # I saw that the function utils.to_dense_batch can achieve the same thing
self.fc = torch.nn.Linear(N * C, 8)
```

Then I should simply be able to run:
`x = self.fc(self.flatten(x))`

Am I doing something inherently wrong?

Posteriorly I run the "inverse" operation to feed to a decoder:
```
self.dfc = torch.nn.Linear(8, N * C)
self.reshape = lambda x : x.reshape(-1, C)

x = self.reshape(self.dfc(x))
```

Thanks and sorry for the silly question

**Comment by rusty1s** [comment]
Is your graph-permutation invariant? Because then it might not be the best idea to input `N * C` features into your linear layer, but instead move the node dimension to the batch dimension and aggregate node features afterwards (using something like `mean`, `sum`, or `max`).

**Comment by aykroyd-technicolor** [comment]
What do you mean by permutation invariant? I don't think so, my dataset comes from a set of meshes in correspondence, and the topology is always shared between data in each layer of my network.

**Comment by rusty1s** [comment]
Ok, so this should theoretically work. Personally, it is hard to judge why your model stops training. Hidden latent dim seems quite low-dimensional though.
